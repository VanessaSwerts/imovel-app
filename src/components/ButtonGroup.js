import React from 'react'
import { View, StyleSheet, TouchableOpacity, Text, FlatList } from 'react-native'

import colors from '../constants/colors.json'

export default function ButtonGroup() {

  const state = {
    data: [
      { id: "0", name: "Casa" },
      { id: "1", name: "Apartamento" },
      { id: "2", name: "Kitnet" },
      { id: "3", name: "SÃ­tio" },
      { id: "4", name: "Studio" },
      { id: "5", name: "Outro" },
    ]
  };

  return (
    <View style={styles.groupContainer} >

      <FlatList
        scrollEnabled={false}
        data={state.data}
        keyExtractor={item => item.id}
        numColumns={3}
        renderItem={({ item }) => {
          return (
            <TouchableOpacity key={item.id} style={styles.button}>
              <Text style={styles.buttonText}>{item.name}</Text>
            </TouchableOpacity>
          )
        }}
      />

    </View>
  )
}

const styles = StyleSheet.create({
  groupContainer: {
    width: "100%",
    marginTop: 10,
    alignItems: 'center',

  },

  button: {
    height: 33,
    width: 96,
    borderRadius: 24,
    borderColor: colors['blue-secondary'],
    borderWidth: 1,

    marginTop: 10,
    marginHorizontal: 4,

    justifyContent: 'center',
    alignItems: 'center',
  },

  buttonText: {
    color: colors['blue-secondary'],
    fontSize: 12,
  }
})