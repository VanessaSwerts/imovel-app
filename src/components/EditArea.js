import React from 'react'
import { View, StyleSheet, Text, TextInput } from 'react-native'

import colors from '../consts/colors.json'

export default function EditArea({ label, text, edit, keyboardType }) {
  return (
    <View style={styles.inputContainer}>
      {
        edit
          ?
          <TextInput style={styles.infoText} keyboardType={keyboardType}
            placeholder={text}
            placeholderTextColor={colors["blue"]}
          />
          : <View style={styles.infoContainer} >
            <Text style={styles.label} >{label}</Text>
            <Text style={styles.infoText} >{text}</Text>
          </View>
      }
    </View>
  )
}

const styles = StyleSheet.create({
  inputContainer: {
    backgroundColor: "#FFF",
    alignSelf: 'stretch',
    height: 45,
    borderWidth: 1,
    borderColor: colors["blue"],
    borderRadius: 24,
    marginTop: 25,
    paddingHorizontal: 12,
    flexDirection: "row",
    shadowOpacity: 0.2,
    shadowRadius: 2,
    shadowOffset: {
      height: 0,
      width: 0,
    },
    elevation: 2,
  },

  infoContainer: {
    paddingHorizontal: 5,
    alignSelf: 'center',
    flexDirection: 'row',
  },

  label: {
    fontWeight: '600',
    color: colors["blue"],
  },

  infoText: {
    color: colors["blue"],
  },

  button: {
    alignSelf: 'center',
    paddingHorizontal: 5,
  },
})