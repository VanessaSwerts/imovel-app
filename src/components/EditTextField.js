import React from 'react'
import { View, StyleSheet, Text, TextInput } from 'react-native'

import colors from '../constants/colors.json'

export default function EditTextField({ label, text, keyboardType }) {
  return (
    <View style={styles.inputContainer}>

      <View style={styles.infoContainer} >
        <Text style={styles.label} >{label}</Text>
        <TextInput style={styles.infoText} keyboardType={keyboardType}
          placeholder={text}
          placeholderTextColor={colors["blue"]}
        />
      </View>

    </View>
  )
}

const styles = StyleSheet.create({
  inputContainer: {
    backgroundColor: "#FFF",
    alignSelf: 'stretch',
    height: 46,
    borderWidth: 1,
    borderColor: colors["blue"],
    borderRadius: 24,
    marginTop: 15,
    paddingHorizontal: 12,
    flexDirection: "row",
    shadowOpacity: 0.2,
    shadowRadius: 2,
    shadowOffset: {
      height: 0,
      width: 0,
    },
    elevation: 2,
  },

  infoContainer: {
    paddingHorizontal: 5,
    alignSelf: 'center',
    flexDirection: 'row',
  },

  label: {
    fontWeight: '600',
    color: colors["blue"],
  },  
})