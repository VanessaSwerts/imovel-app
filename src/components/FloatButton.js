import React from 'react'
import {
  StyleSheet,
  TouchableOpacity
} from 'react-native'
import { FontAwesome5 } from '@expo/vector-icons'

import colors from '../constants/colors.json'

export default function FloatButton(props) {
  return (
    <TouchableOpacity {...props} style={styles.container}>
      <FontAwesome5 name="plus" color="#FFFFFF" size={16} />
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    backgroundColor: colors["blue-secondary"],
    width: 60,
    height: 60,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 560,
    right: 20,
    bottom: 30
  }
})