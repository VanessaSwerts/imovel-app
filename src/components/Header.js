import React from 'react'
import { View, Text, Image, StyleSheet } from 'react-native'

import colors from '../consts/colors.json'

export default function Header({ title }) {


    return (
        <>
            <View style={styles.container}>
                <Image style={styles.logo} source={require('../../assets/img/splash.png')} />
                <Text style={styles.title}>{title}</Text>
            </View>
        </>
    )
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: colors["blue"],
        width: "100%",
        height: "12%",
        minHeight: 70,
        maxHeight: 120,
        padding: 15,
        paddingBottom: 18,
        alignItems: "center",        
        flexDirection: "row",
    },

    title: {
        color: colors["white"],
        fontSize: 24,
        fontWeight: "bold",
        textAlign: "center",
        paddingLeft: 100,
        marginBottom: -15,
    },

    logo: {
        height: 40,
        width: 40,
        marginBottom: -15,
    },
})