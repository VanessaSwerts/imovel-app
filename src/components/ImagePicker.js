import React, { useState, useEffect } from 'react'
import { TouchableOpacity, Image, View, Platform, StyleSheet, Text } from 'react-native'
import * as ImagePicker from 'expo-image-picker'
import Constants from 'expo-constants'

import colors from '../consts/colors.json'


export default function ImagePickerFunction() {
  const [image, setImage] = useState(null)

  useEffect(() => {
    (async () => {
      if (Platform.OS !== 'web') {
        const { status } = await ImagePicker.requestCameraRollPermissionsAsync()
        if (status !== 'granted') {
          alert('Para prosseguir, é necessário permitir o acesso a câmera!')
        }
      }
    })()
  }, [])

  const pickImage = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });

    console.log(result)

    if (!result.cancelled) {
      setImage(result.uri)
    }
  };

  return (
    <View style={styles.containerInput}>
      {image && <Image source={{ uri: image }} style={styles.avatar} />}
      { image
        ? <TouchableOpacity style={styles.button} onPress={pickImage}>
          <Text style={styles.buttonText}>Alterar foto</Text>
        </TouchableOpacity>
        : <TouchableOpacity style={styles.button} onPress={pickImage}>
          <Text style={styles.buttonText}>Adicionar foto</Text>
        </TouchableOpacity>}

    </View>
  );
}


const styles = StyleSheet.create({
  containerInput: {
    alignItems: 'center',
    justifyContent: 'center',
  },

  button: {
    marginTop: 20,
    height: 35,
    width: 150,
    backgroundColor: colors['blue'],
    borderRadius: 24,
    alignItems: 'center',
    justifyContent: 'center',
  },

  buttonText: {
    color: colors['yellow'],
    fontWeight: 'bold',
    fontSize: 16,
  },

  avatar: {
    width: 180,
    height: 180,
    borderRadius: 100,
    borderWidth: 3,
    borderColor: colors["yellow"],
  },


})