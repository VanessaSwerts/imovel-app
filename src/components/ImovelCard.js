import React from 'react'
import { View, Text, Image, StyleSheet, Dimensions, TouchableOpacity, ScrollView } from 'react-native'
import { FontAwesome5 } from '@expo/vector-icons'

import exemplo_house1 from '../../assets/img/exemplo_house1.jpg'
import exemplo_house2 from '../../assets/img/exemplo_house2.jpg'
import colors from '../consts/colors.json'


const CARD_WIDTH = Dimensions.get('window').width * 0.84
const CARD_HEIGHT = Dimensions.get('window').height * 0.27

export default function ImovelCard({ address, city }) {

	listTest = {
		image: [
			{ 'id': 1, 'img': exemplo_house1 },
			{ 'id': 2, 'img': exemplo_house2 },
			{ 'id': 3, 'img': exemplo_house1 },
			{ 'id': 4, 'img': exemplo_house2 },
		]
	}

	return (
		<View style={[styles.card]} >
			<ScrollView
				style={styles.scrollImages}
				pagingEnabled
				decelerationRate={0}
				snapToInterval={CARD_WIDTH}
				snapToAlignment='center'
				showsHorizontalScrollIndicator={true}
				horizontal={true}
			>
				{listTest.image.map((item) => (
					<Image key={item.id} style={styles.images} resizeMode="cover" source={item.img} />
				))}
			</ScrollView>

			<View style={styles.infoContainer} >
				<Text style={styles.type}>Casa</Text>
				<Text style={styles.address} numberOfLines={3}>{address}</Text>
				<Text style={styles.city} numberOfLines={3}>{city}</Text>
				<View style={styles.footerContainer}>					
					<Text style={styles.price}>Aluguel R$: 1.000,00</Text>
					<View style={styles.buttonsContainer}>
						{/* <TouchableOpacity style={styles.button} >
							<FontAwesome5 name={'trash-alt'} size={18} color={colors['blue']} />
						</TouchableOpacity> */}
						<TouchableOpacity style={styles.button}>
							<FontAwesome5 name={'heart'} size={18} color={colors['blue']} />
						</TouchableOpacity>
					</View>
				</View>
			</View>
		</View>
	)
}

const styles = StyleSheet.create({
	card: {
		borderWidth: 1,
		borderColor: '#DDD',
		borderRadius: 24,
		overflow: 'hidden',
		backgroundColor: '#FFF',		
	},

	scrollImages: {
		height: 180,
		maxWidth: '100%',
	},

	images: {
		width: CARD_WIDTH,
		height: CARD_HEIGHT,
	},

	infoContainer: {
		paddingHorizontal: 15,
		paddingVertical: 10,
		borderTopColor: colors['platinum'],
		borderTopWidth: 1.5,
	},

	type: {
		fontSize: 12,
		fontWeight: '300',
		color: "#999",
		marginBottom: 5,
	},

	address: {
		fontSize: 18,
		fontWeight: '600',
		lineHeight: 20,
		color: colors['black'],		
	},

	city: {
		fontSize: 16,
		fontWeight: '400',
		color: colors['black'],
	},

	footerContainer: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',		
	},

	price: {
		fontSize: 14,
		fontWeight: '400',
		color: '#333',
		alignSelf: 'center'
	},

	buttonsContainer: {
		flexDirection: 'row',		
	},

	button: {
		width: 40,
		height: 40,
		borderRadius: 25,
		borderWidth: 1,
		borderColor: colors['platinum'],
		backgroundColor: '#FFF',		
		justifyContent: "center",
		alignItems: "center",
		marginHorizontal: 7,
		elevation: 2,
		shadowColor: '#000',
        shadowOpacity: 0.08,
        shadowRadius: 2,
        shadowOffset: {
            width: 0,
            height: 2,
        }
	}
})