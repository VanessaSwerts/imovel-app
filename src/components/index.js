// Im√≥vel Card
import ImovelCard from './ImovelCard'

// Modal
import ModalContact from './Modal'

// Inputs Components
import FloatButton from './FloatButton'
import InputArea from './InputArea'
import InputSelect from './InputSelect'

// Image Components
import ImagePickerFunction from './ImagePicker'
import SwiperImage from './SwiperImage'

// Navigation Component
import TabMenu from './TabMenu'

export {
    ImovelCard,
    ModalContact,
    FloatButton,
    InputArea,
    InputSelect,
    ImagePickerFunction,
    SwiperImage,
    TabMenu
}
