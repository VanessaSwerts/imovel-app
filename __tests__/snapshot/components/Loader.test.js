import React from 'react'
import { create } from 'react-test-renderer'
import { Loader } from '../../../src/components'

jest.runAllTimers()

describe('Loader snapshot test', () => {
  it('render Loader component correctly when isLoading is false', () => {
    const tree = create(<Loader isLoading={false} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('render Loader component correctly when isLoading is true', () => {
    const tree = create(<Loader isLoading={true} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
