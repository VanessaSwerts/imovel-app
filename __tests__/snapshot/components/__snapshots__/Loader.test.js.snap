// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot test render Loader component correctly when isLoading is false 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      animated={true}
      animationConfig={
        Object {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={true}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={0}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(5, 101, 252, 0.3)",
            "borderColor": "#0565FC",
            "borderRadius": 4,
            "borderWidth": 0,
            "overflow": "hidden",
            "width": null,
          },
          undefined,
        ]
      }
      useNativeDriver={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "#0565FC",
            "height": 5,
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateX": -0,
              },
              Object {
                "scaleX": 0.30007,
              },
            ],
          }
        }
      />
    </View>
  </View>
</Modal>
`;

exports[`Loader snapshot test render Loader component correctly when isLoading is true 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      animated={true}
      animationConfig={
        Object {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={true}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={0}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(5, 101, 252, 0.3)",
            "borderColor": "#0565FC",
            "borderRadius": 4,
            "borderWidth": 0,
            "overflow": "hidden",
            "width": null,
          },
          undefined,
        ]
      }
      useNativeDriver={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "#0565FC",
            "height": 5,
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateX": -0,
              },
              Object {
                "scaleX": 0.30007,
              },
            ],
          }
        }
      />
    </View>
  </View>
</Modal>
`;
